std/array.d

```diff
@@ -413,18 +419,93 @@ if (isInputRange!Range)
     return aa;
 }
 
+/// ditto
+auto assocArray(Keys, Values)(Keys keys, Values values)
+if (isInputRange!Values && isInputRange!Keys)
+{
+    static if (isDynamicArray!Keys && isDynamicArray!Values)
+    {
+        void* aa;
+        {
+            // aaLiteral is nothrow when the destructors don't throw
+            static if (is(typeof(() nothrow
+            {
+                import std.range : ElementType;
+                import std.traits : hasElaborateDestructor;
+                alias KeyElement = ElementType!Keys;
+                static if (hasElaborateDestructor!KeyElement)
+                    KeyElement.init.__xdtor();
+
+                alias ValueElement = ElementType!Values;
+                static if (hasElaborateDestructor!ValueElement)
+                    ValueElement.init.__xdtor();
+            })))
+            {
+                scope(failure) assert(0);
+            }
+            aa = aaLiteral(keys, values);
```

Comment:
If allowing ranges of unequal length, above this should go code along the lines of:
```d
if (keys.length > values.length)
    keys = keys[0 .. values.length];
else if (values.length > keys.length)
    values = values[0 .. keys.length;
```
to avoid an assertion error from `aaLiteral`.